packages/sitecore-jss-nextjs/src/middleware/redirects-middleware.ts

```diff
@@ -285,4 +289,54 @@ export class RedirectsMiddleware extends MiddlewareBase {
     }
     return redirect;
   }
+
+  /**
+   * Generates all possible permutations of an array of key-value pairs.
+   * This is used to create every possible combination of URL query parameters.
+   * @param {Array<[string, string]>} array - The array of key-value pairs to permute.
+   * @returns {Array<Array<[string, string]>>} - A 2D array where each inner array is a unique permutation of the input.
+   */
+  private getPermutations(array: [string, string][]): [string, string][][] {
```

Comment:
I think it makes sense to move this in the base sitecore-jss package. in src/utils/utils.ts .What do you think?

Reply:
Moved